foxpuppet/windows/browser/bookmarks/bookmark.py

```diff
@@ -0,0 +1,218 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+"""Contains classes for handling Firefox bookmarks."""
+
+from selenium.webdriver.common.by import By
+from selenium.common.exceptions import NoSuchElementException
+from selenium.webdriver.remote.webelement import WebElement
+from selenium.webdriver.common.keys import Keys
+from foxpuppet.windows.browser.navbar import NavBar
+from typing import TYPE_CHECKING, Optional, TypedDict, List
+
+
+class BookmarkData(TypedDict):
+    """Bookmark properties."""
+
+    name: str
+    url: str
+    tags: Optional[List[str]]
+    keyword: Optional[str]
+
+
+class BasicBookmark(NavBar):
+    """Handles basic bookmark operations."""
+
+    if TYPE_CHECKING:
+        from foxpuppet.windows.browser.window import BrowserWindow
+
+    @staticmethod
+    def create(window: "BrowserWindow", root: WebElement) -> Optional["BasicBookmark"]:
+        """Create a bookmark object.
+
+        Args:
+            window (:py:class:`BrowserWindow`): Window object this bookmark appears in
+            root (:py:class:`~selenium.webdriver.remote.webelement.WebElement`): WebDriver element object for bookmark
+
+        Returns:
+            :py:class:`BaseBookmark`: Bookmark instance or None
+        """
+        with window.selenium.context(window.selenium.CONTEXT_CHROME):
+            try:
+                return BasicBookmark(window, root)
+            except NoSuchElementException:
+                return None
```

Comment:
I am curious as to why you'd like to return None and not throw the error.

Reply:
I think it might be less verbose. But, now again, it could confusing and unclear. I will just correct that.